### ISSUE TITLE
Headers serialization

### ISSUE THREAD

**Issue by char0n** [swagger-api/swagger-js#1508]
### Q&A (please complete the following information)
 - OS: [e.g. linux]
 - Environment: [e.g. Chrome 59, Node.js v13.3.0]
 - Method of installation: [e.g. npm, unpkg]
 - Swagger-Client version: dev (commit: 0f6e38837b5d247928d9668e6985bd2507ff4f1b)
 - Swagger/OpenAPI version: [*]

### Content & configuration

Swagger-Client usage:
```js
SwaggerClient({
  // your config options here
})
```

### Describe the bug you're encountering

HTTP Client Header serialization algorithm is expected to produce following results when dealing with multiple headers with the same name:

```js
// eg: Cookie: one
//     Cookie: two
//  =  { Cookie: [ "one", "two" ]
```
This will never happen, as the [Headers](https://developer.mozilla.org/en-US/docs/Web/API/Headers/append) interface already merges headers with the same name to one header field and separating them with `, ` characters.

### To reproduce...

1. Create a simple HTTP node server

```js
const http = require('http');

const requestListener = function (req, res) {
  res.setHeader('Content-Type', ['text/plain', 'application/json']);
  res.writeHead(200);
  res.end('Hello, World!');
}
```
2. Save it as `server.js` and run it
```sh
$ node server.js
```

3. create file `test.js` with following content and run it

```js
'use strict';

const SwaggerClient = require('./index');

const request = {
  url: 'http://localhost:8080',
  method: 'GET',
};

SwaggerClient.http(request).then(res => res.headers).then(console.log)
```
```sh
 $ node test.js
```

Result is: 

```js
{
  'content-type': 'text/plain',
  date: 'Wed, 22 Apr 2020 15:52:38 GMT',
  connection: 'close',
  'transfer-encoding': 'chunked'
}
```

### Expected behavior

Result is:
```js
{
  'content-type': [ 'text/plain', 'application/json' ],
  date: 'Wed, 22 Apr 2020 15:52:38 GMT',
  connection: 'close',
  'transfer-encoding': 'chunked'
}
```

### Additional context or thoughts

Native browser [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) and [node-fetch](https://github.com/node-fetch/node-fetch/) are handling duplicate headers differently. I created an [issue](https://github.com/node-fetch/node-fetch/issues/783) on `node-fetch` repository to point out that the behavior of the library is not consistent with the native `fetch` behavior and possibly not compliant with the [specification](https://fetch.spec.whatwg.org/#dom-headers-append).

**Comment by char0n** [comment]
`Content-Type` seems like a special case that needs to be fixed in `node-fetch`. Other headers behave correctly. Observe the POC:

**Server**:
```js
const http = require('http');

const requestListener = function (req, res) {
    res.setHeader('Cookie', ['one=1', 'two=2']);
    res.setHeader('X-Custom-Header', ['value1', 'value2']);
    res.writeHead(200);
    res.end('Hello, World!');
}

const server = http.createServer(requestListener);
server.listen(8080);
```

**Request:**

```js
SwaggerClient = require('swagger-client');

SwaggerClient.http({ url :'http://localhost:8080' }).then(r => r.headers['x-custom-header']); //=> "value1, value2"
```

**Comment by char0n** [comment]
Reducing of `Header.entries()` have been used as primary mechanism for header serialization.

**Comment by char0n** [comment]
Closed by #1523.
